datanode/dehistory/service_test.go

```diff
@@ -376,122 +375,241 @@ func TestMain(t *testing.M) {
 	}
 }
 
-func setupTestSQLMigrations() (int, fs.FS) {
-	sourceMigrationsDir, err := os.Getwd()
-	if err != nil {
-		panic(err)
-	}
+func TestRestoringNodeThatAlreadyContainsData(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	sourceMigrationsDir, _ = filepath.Split(sourceMigrationsDir)
-	sourceMigrationsDir = filepath.Join(sourceMigrationsDir, "sqlstore", "migrations")
+	log := logging.NewTestLogger()
 
-	testMigrationsDir, err = os.MkdirTemp("", "migrations")
+	deHistoryStore.ResetIndex()
+	emptyDatabase()
+
+	snapshotCopyFromPath := t.TempDir()
+	snapshotCopyToPath := t.TempDir()
+	snapshotService := setupSnapshotService(sqlConfig, snapshotCopyFromPath, snapshotCopyToPath)
+
+	ctxWithCancel, cancelFn := context.WithCancel(ctx)
+
+	evtSource := newTestEventSourceWithProtocolUpdateMessage()
+
+	pus := service.NewProtocolUpgrade(nil, log)
+	puh := dehistory.NewProtocolUpgradeHandler(log, pus, func(ctx context.Context, chainID string,
+		toHeight int64,
+	) error {
+		return nil
+	})
+
+	// Run events to height 1800
+
+	broker, err := setupSQLBroker(ctx, sqlConfig, snapshotService,
+		func(ctx context.Context, service *snapshot.Service, chainId string, lastCommittedBlockHeight int64) {
+			if lastCommittedBlockHeight == 1800 {
+				cancelFn()
+			}
+		},
+		evtSource, puh)
 	if err != nil {
-		panic(err)
+		panic(fmt.Errorf("failed to setup broker:%w", err))
```

Comment:
Why are we panicking in tests, and not using `require.NoError(t, err)`?

Reply:
sorted